docs/build/tools/wallets/wallet-libraries/web3Auth.mdx

```diff
@@ -0,0 +1,747 @@
+
+const adapters = await getDefaultExternalAdapters({ options: web3AuthOptions })
+adapters.forEach((adapter: IAdapter<unknown>) => {
+  web3auth.configureAdapter(adapter)
+})
+
+interface UserInfo {
+  email: string;
+  name: string;
+  profileImage: string;
+  [key: string]: unknown;
+}
+
+function App() {
+  const [provider, setProvider] = (useState < IProvider) | (null > null)
+  const [loggedIn, setLoggedIn] = useState < boolean > false
+  const [loading, setLoading] = useState < boolean > true
+  const [address, setAddress] = useState < string > ''
+  const [balance, setBalance] = useState < string > ''
+  const [userInfo, setUserInfo] = (useState < UserInfo) | (null > null)
+
+  useEffect(() => {
+    const init = async () => {
+      try {
+        await web3auth.initModal()
+        setProvider(web3auth.provider)
+
+        if (web3auth.connected) {
+          setLoggedIn(true)
+          await updateUserInfo()
+        }
+      } catch (error) {
+        console.error(error)
+      } finally {
+        setLoading(false)
+      }
+    }
+
+    init()
+  }, [])
+
+  const updateUserInfo = async () => {
+    if (web3auth.provider) {
+      const user = await web3auth.getUserInfo()
+      // @ts-ignore
+      setUserInfo(user)
+      const accounts = await RPC.getAccounts(web3auth.provider)
+      setAddress(accounts)
+      const balance = await RPC.getBalance(web3auth.provider)
+      setBalance(balance)
+    }
+  }
+}
+```
+
+```js etherRPC.ts
+import type { IProvider } from "@web3auth/base";
+import { ethers } from "ethers";
+
+import { contractAddress, contractABI } from "./constants";
+
+const getChainId = async (provider: IProvider): Promise<string> => {
+  try {
+    const ethersProvider = new ethers.BrowserProvider(provider);
+    const networkDetails = await ethersProvider.getNetwork();
+    return networkDetails.chainId.toString();
+  } catch (error) {
+    return error as string;
+  }
+};
+
+const getAccounts = async (provider: IProvider): Promise<string> => {
+  try {
+    const ethersProvider = new ethers.BrowserProvider(provider);
+    const signer = await ethersProvider.getSigner();
+    const address = await signer.getAddress();
+    return address;
+  } catch (error) {
+    return error as string;
+  }
+};
+
+const getBalance = async (provider: IProvider): Promise<string> => {
+  try {
+    const ethersProvider = new ethers.BrowserProvider(provider);
+    const signer = await ethersProvider.getSigner();
+    const address = await signer.getAddress();
+    const balance = ethers.formatEther(await ethersProvider.getBalance(address));
+    return balance;
+  } catch (error) {
+    return error as string;
+  }
+};
+
+const sendTransaction = async (provider: IProvider): Promise<any> => {
+  try {
+    const ethersProvider = new ethers.BrowserProvider(provider);
+    const signer = await ethersProvider.getSigner();
+    const destination = "0x75Bc50a5664657c869Edc0E058d192EeEfD570eb";
+    const amount = ethers.parseEther("1");
```

Comment:
If you want to make a util function, please make it accept the parameters, not constant.

Reply:
If not, I think it doesn't need to be a util function. Looks better to just use this code in the React components.